### ISSUE TITLE
Kubuntu 22.04 LTS: clpeak 1.1.0-rc2 apt version works while clpeak snap version 1.1.2 don't

### ISSUE THREAD

**Issue by lorn10** [krrishnarraj/clpeak#105]
Hi all

Here follows a short bug report about clpeak on **Kubuntu 22.04 LTS**.

When I want to start the snap version of clpeak (1.1.2) I get:
```
clpeak
clGetPlatformIDs (-1001)
no platforms found
```

So no OpenCL environment is detected.

Interestingly the apt version of clpeak (1.1.0-rc2) starts up fine (although it crashes later because of an LLVM or Mesa clover bug).  **Edit**: Maybe that crash is related somehow to LLVM bug [55698](https://github.com/llvm/llvm-project/issues/55698). But that is a different topic.

This happens with stock Ubuntu Mesa as well as with the devel version from the oibaf PPA. Currently installed is: `Mesa 23.2.0-devel (git-1a24f43 2023-05-19 jammy-oibaf-ppa)`.

So at the moment the snap version of clpeak doesn't work. :wink:

**Comment by krrishnarraj** [comment]
The snap system has limited access to filesystem by default due to security reasons. Thats the reason snap version of any opencl app cannot detect all the platforms in the system

**Comment by lorn10** [comment]
Thanks for the fast response. So I had to start it with `sudo`? I didn't try it with higher rights. :disappointed: 

Whatever, - maybe it would make sense to release clpeak also as a (standalone) **AppImage**. On (K)Ubuntu this works perfectly fine also for applications which are way more demanding in regard to the functionality. As an example I can name the Playstation 1 emulator [PCSX-Redux](https://github.com/grumpycoders/pcsx-redux). 

Those AppImages are working fine for me on Kubuntu 20.04 LTS and also Kubuntu 22.04 LTS. The sole limitation is that AppiImages are only easy launchable via the CLI, - which would be perfectly fine for clpeak. Most Desktop Environments like KDE are blocking AppiImages by default. So they cannot be started via a double-click. But as said, for clpeak that wouldn't be a negative aspect.

**Comment by lorn10** [comment]
Okay I have finally found a way to get a working clpeak version **1.1.2** on **Kubuntu 22.04 LTS**. I have simply downloaded and installed the [clpeak deb package](https://launchpad.net/ubuntu/kinetic/amd64/clpeak/1.1.2-1) for **Kubuntu 22.10 "Kinetic"**. :wink: 

Yes that shouldn't be done usually. And if this is done it is strongly recommended to check first all the dependencies. In my case those were all fulfilled so I just downloaded the deb file and installed it with `sudo apt install ./clpeak_1.1.2-1_amd64.deb`. And this worked perfectly fine. 

However, clpeak is still segfaulting because of the above mentioned LLVM bug https://github.com/llvm/llvm-project/issues/55698. Nevertheless it is nice that I was in the end able to find a "noob" way to get the current clpeak version 1.1.2 also on 22.04 LTS. Will close this bug report. 

Maybe clpeak will start to work again on Radeon TeraScale based hardware with the new **rusticl** OpenCL implementation which is not so dependent on LLVM like clover. :+1:
